### ISSUE TITLE
can't attach debugger when doing `flux run` and then `flux job attach` from login node.

### ISSUE THREAD

**Issue by lee218llnl** [flux-framework/flux-core#5603]
When I do a `flux alloc` and then within the allocation do a `flux run MyApp` I can then `flux attach --debug <JobID>` and attach STAT to the application. However, if I don't explicitly `flux alloc`, but rather directly invoke `flux run MyApp` from a login node and then `flux attach --debug <JobID>`, then STAT gets an error when trying to launch its daemons. The session below demonstrates the behavior (where tioga11 is a login node):

```
tioga11{testnewm}24: flux run -N 2 -n 32 -t 30 a.out &
[1] 3718629
tioga11{testnewm}25: tioga29, MPI task 1 of 32 stalling

tioga11{testnewm}25: flux jobs -A | grep testnewm
f2X8PsZb3eXH pdebug   testnewm a.out       R     32      2   8.840s tioga[29-30]
tioga11{testnewm}26: flux job attach --debug f2X8PsZb3eXH &
[2] 3718742
tioga11{testnewm}27: tioga29, MPI task 1 of 32 stalling

tioga11{testnewm}27: ps x | grep 'job attach'
3718629 pts/60   S      0:00 flux-job attach f2X8PsZb3eXH
3718742 pts/60   S      0:00 job attach --debug f2X8PsZb3eXH
3718820 pts/60   S+     0:00 grep --color=auto job attach
tioga11{testnewm}28: stat-cl 3718742

(STATmain.py:3718889): dbind-WARNING **: 16:11:11.874: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-SDIG3OSUwY: Connection refused
STAT started at 2023-11-29-16:11:11
Attaching to job launcher (null):3718742 and launching tool daemons...
<Nov 29 16:11:11> <Launchmon> (INFO): Just continued the RM process out of the first trap
flux-exec: Error: rank 18: /collab/usr/global/tools/stat/toss_4_x86_64_ib/spack3/opt/spack/linux-rhel8-ivybridge/gcc-8.5.0/stat-4.2.1-mqkguv2cfenhe2xtn7yooes4klb72d7o/bin/STATD: Request requires owner credentials
flux-exec: Error: rank 19: /collab/usr/global/tools/stat/toss_4_x86_64_ib/spack3/opt/spack/linux-rhel8-ivybridge/gcc-8.5.0/stat-4.2.1-mqkguv2cfenhe2xtn7yooes4klb72d7o/bin/STATD: Request requires owner credentials
```

Is there any way to attach a debugger to a job that is run via `flux run` on a login node?

**Comment by garlick** [comment]
Is STAT trying to use `flux exec` to launch tool daemons?  `flux exec` only works if you're the "instance owner", which is the case for the instance of flux started for alloc/batch, but not the case when the instance is the system one.

If we need that capability then we might need to dig a bit deeper to figure out another way to start the tool daemons along side the user's job.

**Comment by lee218llnl** [comment]
yes, STAT, via LaunchMON, uses `flux exec` to launch daemons. I think this would be a nice feature to have. I'm not sure how many of our users run in this manner, and then of course, only a small subset of those would need debugging capabilities, so I appreciate that this may not be high priority.

**Comment by garlick** [comment]
OK, thanks for that context.

One solution might be to embed the Flux subprocess server in the flux shell.  This was prototyped in #2298.  I think there has since been some work in the flux subprocess server that might make this a bit easier were we to try again.

Something to revisit perhaps.

**Comment by grondo** [comment]
That might be nice. Maybe `flux exec` (or an alternative command) could take a `--jobid` option and automatically determine the subprocess server endpoint and launch processes on all shell ranks for ease of use.

In the meantime, could STAT/LaunchMON use `ssh` to handle this case?

**Comment by grondo** [comment]
> the hope is to bulk launch the daemons

You could use `pdsh` which is similar to `flux exec`, but understood. The subprocess server in the job shell has been brought up before, so @garlick's idea is likely the best solution.

**Comment by grondo** [comment]
This should work now since #7430 was merged. You do have to use the `flux exec --jobid` option to target the subprocess server running in the job shell, and to enable payload signatures in a multi-user instance.
